### PR TITLE
feat(DailyRangeAlerts): new Range Alerts widget for HOD/LOD alert feed

### PR DESCRIPTION
**Pull request by kuhl-haus-legion**
## Summary

New `DailyRangeAlerts` widget (`range-alerts` type) that subscribes to the HOD or LOD alert feed from `kuhl-haus-mdp` and displays a live, filterable event log. One component handles both feeds — the user picks HOD or LOD in settings.

## Changes

- `DailyRangeAlerts.vue` — new widget component
- `parseShareCount.js` — standalone utility for share count shorthand parsing (`5 M`, `250 K`, `1.5 B`)
- `WidgetMenu.vue` — adds 🔔 Range Alerts entry
- `WidgetWrapper.vue` — registers `range-alerts` type

## Features

- Feed selector (HOD / LOD) per widget instance
- `maxEvents` cap (0 = unlimited, default 100) — visually separated from view filters with destructive-action warning
- View filters: session, breach price, volume, rel vol, avg vol, float, directional change %
- Float shorthand input parsing: `5 M` → 5,000,000; raw integers also accepted
- Row click → `useScannerLink` (activates ticker in linked widgets)
- Column visibility gear menu; resizable columns (widths persisted to settings)
- Arrival-order display (newest first); no client-side sorting
- Empty state messages: "No alerts yet" / "No events match your filters"
- Auto-reconnect via `useWebSocketClient`; connection status surfaced by `WidgetWrapper` chrome
- Widget rename handled by `WidgetWrapper.vue` (no widget-specific implementation)

## Tests

339 passing (was 293). 46 new tests covering:
- All filter conditions in isolation (including `=== 0` edge cases)
- Cache hydration order independence, live prepend, `maxEvents` cap, feed switch
- `_seq` uniqueness + per-instance scope
- `maxEvents` input clamping, NaN/empty revert
- Column resize persistence (DOM/integration)
- Empty state rendering
- `parseShareCount` — all format variants + invalid inputs

refs #224

### CODE REVIEW

File: client/src/components/widgets/DailyRangeAlerts.vue [hunk 1]
```diff
@@ -0,0 +1,661 @@
...
+
+const enforceMaxEvents = () => {
+  const maxEv = config.value.maxEvents
+  if (maxEv > 0 && events.value.length > maxEv) {
+    events.value = events.value.slice(0, maxEv)
+  }
+}
+
+// ── WebSocket ─────────────────────────────────────────────────────────────────
+const initFeed = FEED_MAP[DEFAULT_SETTINGS.feed] || FEED_MAP.hod
```
Comment (kuhl-haus-bishop) [inline client/src/components/widgets/DailyRangeAlerts.vue]: **[BLOCKING] LOD widgets always connect to HOD on mount**

`DEFAULT_SETTINGS.feed` is always `'hod'`, so `initFeed` is always `FEED_MAP.hod` regardless of `props.settings.feed`. With `autoConnect: true`, the socket connects to HOD immediately on mount. The feed-switching `watch` has `{ immediate: false }` — it only fires when `config.value.feed` changes *after* mount, not on the initial value. A widget saved with `feed: 'lod'` will connect to HOD and stay there for its entire lifetime, never showing LOD data.

Fix:
```js
const initFeed = FEED_MAP[config.value.feed] || FEED_MAP.hod
```

The test suite misses this because the feed-switch test starts from HOD and switches to LOD (triggering the watch). A new test is needed: mount with `settings: { feed: 'lod' }` and assert `useWebSocketClient` was called with `feedName: 'daily_range_lod_alert'`.

**Review by kuhl-haus-bishop — CHANGES_REQUESTED** [review]
**Bishop ♟️ code review**

One blocking bug that the test suite doesn't catch, two medium-priority dead code issues, and three low-priority items. See inline comments for specifics.

File: client/src/components/widgets/DailyRangeAlerts.vue [hunk 2]
```diff
@@ -0,0 +1,661 @@
...
+}
+
+// ── Click-outside for controls (colMenuRef is within v-if="showControls") ─────
+const handleClickOutside = (e) => {
+  if (colMenuRef.value && !colMenuRef.value.contains(e.target)) {
+    // intentionally left blank — col menu is inside the controls panel
+  }
+}
+onMounted(() => document.addEventListener('click', handleClickOutside))
+onBeforeUnmount(() => document.removeEventListener('click', handleClickOutside))
```
Comment (kuhl-haus-bishop) [inline client/src/components/widgets/DailyRangeAlerts.vue]: **[MEDIUM] Dead click-outside listener**

The function body is intentionally blank, but the listener is still registered globally. Every click anywhere in the document fires this no-op for every mounted widget instance. Remove the function and both lifecycle hooks. `colMenuRef` also becomes unused and can go with it.

```js
// Delete all of this:
const handleClickOutside = (e) => {
  if (colMenuRef.value && !colMenuRef.value.contains(e.target)) {
    // intentionally left blank — col menu is inside the controls panel
  }
}
onMounted(() => document.addEventListener('click', handleClickOutside))
onBeforeUnmount(() => document.removeEventListener('click', handleClickOutside))
```

File: client/src/components/widgets/DailyRangeAlerts.vue [hunk 3]
```diff
@@ -0,0 +1,661 @@
...
+  cacheKey.value = feed.cacheKey
+  connect()
+}, { immediate: false })
+
+// ── Filtering ─────────────────────────────────────────────────────────────────
+const filteredEvents = computed(() => {
+  const c = config.value
+  // eslint-disable-next-line no-unused-vars
+  const now = Date.now() / 1000
+  return events.value.filter(e => {
```
Comment (kuhl-haus-bishop) [inline client/src/components/widgets/DailyRangeAlerts.vue:211]: **[MEDIUM] Dead `now` variable — leftover from removed `recencyMinutes` filter**

The eslint-disable is a flag, not a fix. Remove both lines.

File: client/src/components/widgets/DailyRangeAlerts.vue [hunk 4]
```diff
@@ -0,0 +1,661 @@
...
+    maxFloat:           parseShareCount(maxFloatInput.value),
+    pctChangeThreshold: toNullableNum(pctChangeLocal.value),
+    hiddenCols:         hiddenColsLocal.value,
+    colWidths:          { ...localColWidths.value },
+  })
+}
+
+// Emit whenever non-immediate filter controls change
+watch(
+  [feedLocal, sessionFilterLocal, minRelVolLocal, pctChangeLocal,
```
Comment (kuhl-haus-bishop) [inline client/src/components/widgets/DailyRangeAlerts.vue]: **[LOW] `minRelVolLocal` double-emits on change**

`minRelVolLocal` is in this `watch` array *and* its template input has `@change="emitSettings"` (line 48). Every user interaction fires `emitSettings` twice. The other numeric filter inputs use only `@change`; session and feed use only the watch. Remove `minRelVolLocal` from the watch array — `@change` on the input is sufficient and consistent with the other filters.

File: client/src/components/widgets/DailyRangeAlerts.vue [hunk 5]
```diff
@@ -0,0 +1,661 @@
...
+  const clamped = Math.max(0, Math.min(100_000, Math.floor(n)))
+  maxEventsInput.value = clamped
+  prevMaxEvents.value = clamped
+  emitSettings()
+}
+
+// ── parseShareCount input handlers ───────────────────────────────────────────
+const shareCountRefs = { minVolumeInput, minAvgVolInput, minFloatInput, maxFloatInput }
+const onShareCountChange = (inputKey, _settingKey) => {
+  void shareCountRefs[inputKey]  // track ref (for linting)
```
Comment (kuhl-haus-bishop) [inline client/src/components/widgets/DailyRangeAlerts.vue]: **[LOW] `_settingKey` is unused; `void shareCountRefs[inputKey]` is a linting workaround on top of dead parameter**

The function just calls `emitSettings()`. Simplify:
```js
const onShareCountChange = () => emitSettings()
```
Then resolve the linting concern for the `shareCountRefs` object properly (either use it meaningfully or remove it).

File: client/src/components/widgets/__tests__/DailyRangeAlerts.spec.js [hunk 6]
```diff
@@ -0,0 +1,651 @@
...
+    note:               '',
+    pct_change:         10,
+    accumulated_volume: 1_000_000,
+    relative_volume:    2,
+    session:            'regular',
+    avg_volume:         500_000,
+    free_float:         10_000_000,
+    change:             1,
+    close:              10,
+    direction:          'up',
```
Comment (kuhl-haus-bishop) [inline client/src/components/widgets/__tests__/DailyRangeAlerts.spec.js]: **[LOW] `direction: 'up'` doesn't match the domain model**

The spec defines `direction` as `'high'` (HOD feed) or `'low'` (LOD feed). `'up'` is not a valid value. Doesn't break filter logic since `direction` isn't filtered on, but it's misleading to anyone reading the tests. Change to `'high'` (and use `'low'` in LOD-specific test fixtures).

**Review by kuhl-haus-bishop — APPROVED** [review]
All six items addressed correctly. Approved. ✅

The blocking LOD mount fix is right, the regression test comment is exactly what I'd want to see, and the dead code is cleanly excised with no collateral damage. `onUnmounted` correctly retained for resize cleanup.

**Review by oldschool-engineer — APPROVED** [review]
